### ISSUE TITLE
Support for actual int64 as opposed to int53

### ISSUE THREAD

**Issue by rochdev** [darrachequesne/notepack#21]
I was wondering if support for in64 libraries like `long` and `int64-buffer` would be considered. I'm willing to open a PR for `int64-buffer` since it's what we use and is also used by another msgpack library.

This would not add a lot of overhead to the library and would look something like this:

```js
if (value._isUint64BE) {
  value = value.toArray();
  bytes.push(0xcf, value[0], value[1], value[2], value[3], value[4], value[5], value[6], value[7]);
  return 9;
}
```

**Comment by rochdev** [comment]
@darrachequesne Thoughts?

**Comment by slavaGanzin** [comment]
@rochdev create a PR. It's always easier to talk about working code.

p.s. I have no relation to maintainer.

**Comment by rochdev** [comment]
Opened #22

**Comment by darrachequesne** [comment]
That's an interesting idea! My only concern is the impact on the bundle size, for the browser build. Would custom encoders (suggested here: https://github.com/darrachequesne/notepack/issues/18) help?

**Comment by rochdev** [comment]
Good point. I'll have to check if we could use something like that with our requirements (which are a bit complicated) and will report back. Having an external implementation for custom encoding would definitely remove a lot of the decisions related to built-in support.
